### PR TITLE
🏗️ Persist the session

### PR DESCRIPTION
**Pull request by falexwolf**
- https://github.com/laminlabs/lndb-setup/pull/99

### CODE REVIEW

File: lamindb/dev/db/_core.py [hunk 1]
```diff
@@ -1,10 +1,20 @@
+import os
+
 import sqlmodel as sqm
 from lndb_setup import settings
 
+_session = None
+if "LAMIN_SKIP_MIGRATION" not in os.environ:
```
Comment (falexwolf) [inline lamindb/dev/db/_core.py:7]: @fredericenard, I'm using your env for detecting the rest.

**Comment by Koncopd** [comment]
Every call of `settings.instance.db_engine()` synchronizes the local sqlite file with cloud. If the session is stored, then this synchronization doesn't happen anymore.

**Comment by falexwolf** [comment]
Ahhh, thanks for the hint! I moved this logic over to lndb-setup and I'll think what to do about this!

File: lamindb/dev/db/_core.py (hunk 1 above)
Comment (falexwolf) [reply lamindb/dev/db/_core.py]: This is now in lndb-setup!

**Comment by falexwolf** [comment]
My current thought is that the main thing this means is that we need to mutual exclusion check for the sqlite file. And we might need more control for influencing synching for the user.

**Comment by Koncopd** [comment]
The problem with storing the session is that the db file is synchronized only once on session initialization. Actually, this change might break a lot of things. For example, if the cached db file is deleted, it won't be synchronized anymore as far as i understand.

**Comment by falexwolf** [comment]
Yes, I see. It’s an experimental change that’s deeply grounded in how users should experience lazyily loading sqlalchemy objects.

**Comment by falexwolf** [comment]
I’ll walk you through it. And if there is another solution, then that’d be great.

**Comment by falexwolf** [comment]
To note: In the current implementation, the old behavior (function-scoped sessions) is just an environment variable away. So, we can just switch it back anytime.

**Comment by Koncopd** [comment]
What happens with the stored session when the cached db file is deleted? Or replaced? Maybe it is bettter to initiate sync of yhe db file on every session() call.

**Comment by falexwolf** [comment]
We can do that, with some caveats; let’s try scoping both the session and the cache check with the `InstanceSession` lifetime (which is essentially until call of `lndb_setup.init`) for one day or two, and then talk about mutual exclusion and updates in the sqlite case in depth

**Comment by Koncopd** [comment]
Mutual exclusion is not meaningful in this setting, because we don't get any changes from the cloud until new init call where new session is received. 
I mean this change seems to break the sync logic totally.

**Pull request merged**